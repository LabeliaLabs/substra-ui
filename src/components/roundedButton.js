import React from 'react';
import PropTypes from 'prop-types';
import {css} from 'emotion';
import styled from '@emotion/styled';
import {blueGrey, ice, slate} from '../variables/colors';
import {spacingExtraSmall, spacingNormal} from '../variables/spacing';

export const Button = styled.button`
    color: ${slate};
    height: 30px;
    line-height: 28px;
    border-radius: 15px;
    border: 1px solid ${ice};
    padding: 0 ${spacingNormal};
    background: none;
    cursor: pointer;
    
    &:not(:disabled):hover {
        background-color: ${ice};
        transition: background-color 200ms ease-out;
    }
    
    &:disabled {
        color: ${blueGrey}
    }
`;

const icon = css`
    margin-right: ${spacingExtraSmall};
    margin-bottom: -3px;
`;

export const RoundedButton = ({
    disabled, Icon, Button, iconColor, iconColorDisabled, children, ...props
}) => (
    <Button disabled={disabled} {...props}>
        {Icon && (
            <Icon
                width={15}
                height={15}
                color={disabled ? iconColorDisabled : iconColor}
                className={icon}
            />
        )}
        {children}
    </Button>
);

RoundedButton.propTypes = {
    disabled: PropTypes.bool,
    iconColor: PropTypes.string,
    iconColorDisabled: PropTypes.string,
    Icon: PropTypes.element,
    children: PropTypes.node,
    Button: PropTypes.element,
};


RoundedButton.defaultProps = {
    disabled: false,
    iconColor: blueGrey,
    iconColorDisabled: slate,
    Icon: null,
    children: null,
    Button,
};
