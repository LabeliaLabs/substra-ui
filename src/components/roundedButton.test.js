import React from 'react';
import {RoundedButton} from "./roundedButton";
import renderer from 'react-test-renderer';


test('render an actual button', () => {
    const component = renderer.create(
        <RoundedButton/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

