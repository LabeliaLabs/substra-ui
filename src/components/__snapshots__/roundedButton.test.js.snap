// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render an actual button 1`] = `
<button
  className="css-10hh0l8"
  disabled={false}
/>
`;
